Kernel/Memory/MemoryManager.cpp

```diff
@@ -577,6 +582,26 @@ UNMAP_AFTER_INIT void MemoryManager::initialize_physical_pages()
     });
 }
 
+#ifdef HAS_ADDRESS_SANITIZER
+void MemoryManager::initialize_kasan_shadow_memory()
+{
+    m_global_data.with([&](auto& global_data) {
+        // We map every 8 bytes of normal memory to 1 byte of shadow memory, so we need a 1/9 of total memory for the shadow memory.
+        auto virtual_range = global_data.region_tree.total_range();
+        auto shadow_range_size = MUST(page_round_up(ceil_div(virtual_range.size(), 9ul)));
+        dbgln("MM: Reserving {} bytes for KASAN shadow memory", shadow_range_size);
+
+        auto vmobject = MUST(AnonymousVMObject::try_create_with_size(shadow_range_size, AllocationStrategy::AllocateNow));
+        auto& shadow_region = *MUST(Region::create_unplaced(move(vmobject), 0, {}, Memory::Region::Access::ReadWrite)).leak_ptr();
```

Comment:
`*foo.leak_ptr()` feels pretty gross, that's like an owning reference. can we at least keep it as an auto* if we're going to leak the ASAN memory at runtime?

Reply:
sure, changed it to auto*